jenkins/bokchoy_db_pull_request.py

```diff
@@ -0,0 +1,212 @@
+        if repo.name == repo_name:
+            repository = repo
+            break
+    if not repository:
+        raise StandardError(
+            "Could not connect to the repository: {}. "
+            "Please make sure you are using the correct "
+            "credentials and try again.".format(repo_name)
+        )
+    return repository
+
+
+def _get_file_sha(repository, file_path):
+    """
+    Finds the sha of a specific file in the repository.
+    Returns the file sha.
+    """
+    try:
+        # Get the blob sha of the db file on our branch
+        file_sha = repository.get_file_contents(file_path).sha
+    except:
+        raise StandardError(
+            "Could not locate file: {}".format(file_path)
+        )
+    return file_sha
+
+
+def _read_file_contents(file_path):
+    """
+    Read the contents of a file and return a string of the data.
+    Returns a string of the file contents.
+    """
+    try:
+        with open(file_path, 'r') as opened_file:
+            data = opened_file.read()
+    except:
+        raise StandardError(
+            "Unable to read file: {}".format(file_path)
+        )
+    return data
+
+
+def _file_has_changed(repository, local_file, file_path):
+    """
+    Checks to see if a file differs from the current version
+    in a repo.
+    Returns true if the files are different.
+    """
+    try:
+        repo_file_contents = repository.get_file_contents(file_path).decoded_content
+    except:
+        raise StandardError(
+            "Could not get contents of file: {}".format(file_path)
+        )
+    return local_file != repo_file_contents
+
+
+def _create_branch(repository, branch_name, sha):
+    """
+    Create a new branch with the given sha as its head.
+    """
+    try:
+        repository.create_git_ref(branch_name, sha)
+    except:
+        raise StandardError(
+            "Unable to create git branch: {}. "
+            "Check to make sure this branch doesn't already exist.".format(branch_name)
+        )
+
+
+def _update_file(repository, file_path, commit_message, contents, file_sha, branch_name):
+    """
+    Create a commit on a branch that updates the file_path with the string contents.
+    """
+    try:
+        repository.update_file(file_path, commit_message, contents, file_sha, branch_name)
+    except:
+        raise StandardError(
+            "Error updating database file: {}".format(file_path)
+        )
+
+
+def _create_pull_request(repository, title, body, base, head):
+    """
+    Create a new pull request with the changes in head.
+    """
+    try:
+        pull_request = repository.create_pull(
+            title=title,
+            body=body,
+            base=base,
+            head=head
+        )
+    except:
+        raise StandardError(
+            "Could not create pull request"
+        )
+
+
+@click.command()
+@click.option(
+    '--sha',
+    help="Sha of the merge commit to base the new PR off of",
+    required=True,
+)
+@click.option(
```

Comment:
The token will be present as an environment variable (via credentials binding). Should we just access it via os.environ?  That way it won't be present in the console log. What do you think?